### PR TITLE
added ConnEstablished failed event, some cleanup

### PR DESCRIPTION
**Pull request by rwlockbg**

### CODE REVIEW

File: crates/net/network/src/frost/protocol.rs [hunk 1]
```diff
@@ -109,14 +109,14 @@ impl ConnectionHandler for FrostConnectionHandler {
...
-            })
-            .ok();
+        let connection_established_event = FrostProtocolEvent::ConnectionEstablished {
+            direction,
+            peer_id,
+            to_connection: remote_peer_tx,
+        };
+        if let Err(e) = self.state.events.send(connection_established_event) {
+            error!(target: "network::frost::protocol::into_connection", "Failed to send ConnectionEstablished event: {:?}", e);
+        }
```
Comment (arminsabouri) [inline crates/net/network/src/frost/protocol.rs:119]: Do we want to return with an error here? or just log?

**Review by arminsabouri — APPROVED** [review]

File: crates/net/network/src/frost/protocol.rs (hunk 1 above)
Comment (rwlockbg) [reply crates/net/network/src/frost/protocol.rs:119]: This is the the most important event, I would say yes, it needs to be an error as this is place in code where the rlxp protocol makes the durable connection to the other peer.

Comment (arminsabouri) [reply crates/net/network/src/frost/protocol.rs:119]: Will we violate the trait if we return an Result here?

Comment (arminsabouri) [reply crates/net/network/src/frost/protocol.rs:119]: Lets merge and open a ticket for this. Will be good to just start with some logging

**Pull request merged**